src/functions.php

```diff
@@ -344,3 +346,47 @@ function _current_time()
 {
     return function_exists('hrtime') ? hrtime(true) / 1e9 : microtime(true);
 }
+
+
+/**
+ * @param int $options
+ *
+ * @return UriInterface
+ *
+ * @internal
+ */
+function _idn_uri_convert(UriInterface $uri, $options = 0)
+{
+    if ($uri->getHost()) {
+        $variant = defined('INTL_IDNA_VARIANT_UTS46') ? INTL_IDNA_VARIANT_UTS46 : 0;
+        $asciiHost = idn_to_ascii($uri->getHost(), $options, $variant, $info);
```

Comment:
Just checked locally, it raises a deprecation error (`idn_to_ascii(): INTL_IDNA_VARIANT_2003 is deprecated`).

The only way is to call the function without the third argument at all, like [here](https://github.com/TYPO3/TYPO3.CMS/commit/3a2b54a1b36b99b525cc04eccf20fd018a24e4a6#diff-cb4f296a39c2f11c2784e0e940745b82R193-R199):
```
$asciiHost = defined('INTL_IDNA_VARIANT_UTS46')
                ? idn_to_ascii($uri->getHost(), $idnOptions, INTL_IDNA_VARIANT_UTS46, $info)
                : idn_to_ascii($uri->getHost(), $idnOptions);
```

Reply:
thanks again..  done..

Reply:
OK, so it's night and too many versions to keep in the head.

It works on my machine, because I have PHP 7.4 with already right defaults.

On 7.3 it raises a warning by default, if you don't specify `INTL_IDNA_VARIANT_UTS46` manually. So there is basically no way to get rid of this deprecation error in PHP 7.2+ if you have an old version of ICU library. Looks like we have to disable `idn_conversion` by default in this case (PHP 7.2+ and `INTL_IDNA_VARIANT_UTS46` not defined).

Reply:
@gmponos, now I think it's better to open a separate PR for this change. Please take a look at #2453.

Reply:
better we avoid the back and forth :)

Reply:
Agree, but now it seems really out of scope here